### PR TITLE
[WT-444] Repurpose Concessions to Functioning Fees

### LINKED ISSUE #444: Ticketing - Concessions Tickets should be repurposed for Fees
When attempting to purchase a concessions ticket, the functionality is inconsistent and does not show up properly in Stripe. Teams should communicate with Portland Playhouse and/or Ben to understand how this feature should work and fix it.

### PR DESCRIPTION
**Pull request by henrykaus**
### Description
Repurposed Concessions for Fees across the following tables:
* `tickettypes`
* `ticketrestrictions`
* `seasontickettypepricedefaults`

Fees have the following functionality:
* An admin can set a fee price at the ticket type, season or showing level.
* Upon purchase of a ticket with an associated fee, that fee is tacked onto the order for each number of ticket of that showing. So if you to buy `4` GA tickets to a particular 8 o'clock showing, you would be charged `4` of those fees.
* If the ticket has a price of `$0`, then no fee will be charged
  * Although if a discount is used that brings the price down to `$0`, it still charges the fee. Currently unsure what the functionality with fees should be if discounts are used.
* The fee total is tracked with the order in the `orders` and `orderticketitems` table, and is then displayed in the contact order history view.

image

* Also bundled into this PR is an update to the server eslint config to better match the client (in addition to a server-based `.prettierrc` file). This also means that the indentation requirement in the server is now consistently `2 spaces`, rather than a weird mix of 2 and 4. 

### Risks

Refunds do refund the entire amount including the fee amount, however, the fee amount isn't displayed on our refund panel, should this be included?

Unsure around the verbiage for displaying fees to customers. At the moment, I have a tooltip in the checkout process and a description for the line item in Stripe. Not sure if these suffice.

Also like I wrote earlier, not sure if we should still charge fees if discounts subtract from part of the order, or how much in fees we should charge.

image
image

### Validation
Manual testing of Admin Purchase, Admin Reader Purchase, and customer purchase.

### Issue
closes #444 

### Operating System
macOS M2

### CODE REVIEW

File: server/src/models/TicketType.ts
Comment (henrykaus) [inline server/src/models/TicketType.ts:1]: This file was unused (my guess was that its intention was for API v1)

File: client/src/components/Ticketing/ticketingmanager/AdminPurchase/AdminCart.tsx [hunk 1]
```diff
@@ -2,7 +2,8 @@ import React, {ReactElement, useState} from 'react';
 import {useLocation, useNavigate} from 'react-router';
 import {EventRow} from './utils/adminCommon';
 import {getDiscountCode} from './utils/adminApiRequests';
-import {SearchIcon, XIcon} from '../../Icons';
+import {HelpIcon, SearchIcon, XIcon} from '../../Icons';
+import {Tooltip} from '@mui/material';
```
Comment (benfaustin95) [inline client/src/components/Ticketing/ticketingmanager/AdminPurchase/AdminCart.tsx:6]: Should the Tooltip/HelpIcon be added to fee total in the admin cart as well?

**Comment by benfaustin95** [comment]
Do we know what PPH currently does with regards to the tacked on fee when they process a refund? I imagine it could be non-refundable given it is a cost that is incurred at the time of payment processing and not necessarily returned from Stripe if the order is refunded.  If it does end up that we want to include the fee when processing a refund it might be worth storing the fee in the order item price or adding a field to the order item table so that we can refund the fee on an item by item basis.

File: client/src/components/Ticketing/ticketingmanager/AdminPurchase/AdminCart.tsx (hunk 1 above)
Comment (henrykaus) [reply client/src/components/Ticketing/ticketingmanager/AdminPurchase/AdminCart.tsx:6]: I had done that at first, but then realized that admin don't really need a fee disclaimer. I'll remove those stale imports.

File: server/src/controllers/eventController.service.ts [hunk 2]
```diff
@@ -182,6 +197,11 @@ export const getTicketItems = async (
           `Showing ${item.product_id} for ${item.name} does not exist`,
       );
     }
+    const ticketRestriction = eventInstance.ticketRestrictionMap.get(item.typeID);
```
[the remark this answers is not in the document]
Comment (henrykaus) [reply server/src/controllers/eventController.service.ts:257]: I removed the check on 264.

File: server/prisma/migrations/20240319014147_renamed_consessions_to_fee/migration.sql [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+ALTER TABLE "seasontickettypepricedefault" DROP COLUMN "concessionprice",
+ADD COLUMN     "fee" MONEY NOT NULL DEFAULT 0;
+
+-- AlterTable
+ALTER TABLE "ticketrestrictions" DROP COLUMN "concessionprice",
+ADD COLUMN     "fee" MONEY NOT NULL DEFAULT 0;
+
+-- AlterTable
+ALTER TABLE "tickettype" DROP COLUMN "concessions",
+ADD COLUMN     "fee" MONEY NOT NULL;
```
[the remark this answers is not in the document]
Comment (henrykaus) [reply server/prisma/migrations/20240319014147_renamed_consessions_to_fee/migration.sql]: It probably doesn't hurt.

File: client/src/components/Ticketing/ticketingmanager/AdminPurchase/AdminPurchase.tsx [hunk 4]
```diff
@@ -237,6 +240,7 @@ const AdminPurchase = () => {
           ...row,
           complimentary: isChecked,
           price: isChecked ? 0 : row.price,
+          fee: isChecked || row.price === 0 ? 0 : row.fee,
```
[the remark this answers is not in the document]
Comment (henrykaus) [reply client/src/components/Ticketing/ticketingmanager/AdminPurchase/AdminPurchase.tsx:247]: Fixed

**Comment by henrykaus** [comment]
> It looks good, it is cool that fees finally having meaning! With regards to refunding fees do we know what PPH currently does with the tacked on fee when they process a refund? I imagine it could be non-refundable given it is a cost that is incurred at the time of payment processing and not necessarily returned from Stripe if the order is refunded. If it does end up that we want to include the fee when processing a refund it might be worth storing the fee in the order item price or adding a field to the order item table so that we can refund the fee on an item by item basis.

This is a good idea. I added it in. I didn't change much in the way of displaying the individual fees or change anything in the way of refunds since I imagine we'll do that at a later date.

**Review by benfaustin95 — APPROVED** [review]

**Pull request merged**